### ISSUE TITLE
Open source Grouphone and Conqueror

### ISSUE THREAD

**Issue by kaustavdm** [applait/meta#18]
## Proposal

Open source all the technical assets of Grouphone under a custom semi-permissive, non-commercial license. This includes:
- grouphone
- grouphone-api
- conqueror
- ~~graphic assets~~ #20 
## Key points of the license
- Non-commercial usage allowed
- Usage as part of a commercial solution, whether derivative or non-derivative, is allowed only under explicit commercial licensing
- Commercial, non-derivative usage allowed only under explicit commercial license.
- No trademark usage without explicit permission.
- Derivatives cannot be commercial without explicit permission from us
- Non-commercial derivatives allowed
- Derivatives, whether commercial or non-commercial need to provide attribution to this project.
- Derivatives cannot use the same name and graphic assets and branding.
- Patent license granting and copyright granting for non-commercial usage
- Any contributions to this project becomes the copyright of Applait.
## Rationale

Applait wants to revolutionize communication technologies. But, given our political and philosophical standpoint, we will not be content with playing by the rules, and keeping everything in a closed environment.

We need to open up.

But we need to be careful so that we are not marked as radicals. We need to be in the good books of the powerful ones. We need to steer clear of being choked down.

And we need to earn at the same time.

We, also, should pose ourselves as path makers and not just blind followers of trends. That is how we can make a mark in the long run, and avoid getting lost in the ever growing list of startups.

So, we will actively apply for patents and trademarks wherever applicable, in order to keep the big corporations from monopolizing them. Then, we grant exclusive rights to all non-commercial projects. This will foster research and adoption of those technologies.

Any commercial entity has to pay us for using these. So we still earn from them.

We get to show everyone that our code is out there in the open. If others try to use our code for their purposes, that is a win for us. We can sell that to our b2b customers.

The transfer of copyright of external contributions ensures that we own the legal right to make everything available for non-commercial use.

Additionally, this solves the issue about gathering feedback on each project.

In short, we break the entire patenting and copyrights system by exploiting it.
## Blockers
- [ ] grouphone-api has list of emails in its history. They can be removed by removing those commits from the repo and force updating the history
- [ ] Need to clean up existing known security issues.
- [ ] Need to document code properly.
- [ ] Need to cleanup code to make it look better. Something that we can be proud of.
